api_tests/tests/lightning_routes.ts

```diff
@@ -131,4 +132,21 @@ describe("Lightning routes tests", () => {
             ).rejects.toThrow("Invalid body.");
         })
     );
+
+    // This test could be anywhere, it is here because this is the first test file that uses the new comunication protocols.
+    it(
+        "alice-cannot-create-two-identical-swaps",
+        twoActorTest(async ({ alice, bob }) => {
+            const bodies = (await SwapFactory.newSwap(alice, bob))
+                .hanEthereumEtherHalightLightningBitcoin;
+
+            await alice.createSwap(bodies.alice);
+            await sleep(200);
+            try {
+                await alice.createSwap(bodies.bob);
+            } catch (err) {
+                expect(err.status).toBe(400);
+            }
```

Comment:
I personally find `try-catch` to be quite ugly so I usually do this:

```suggestion
			await expect(alice.createSwap(bodies.bob)).toThrow("Swap already exists.");
```

How does this work?

`createSwap` returns a Promise, which is equivalent to a Future from Rust.
`await` does the same thing except that it throws an exception if the Promise _rejects_. You can think of Promises as Futures that always return a Result.

So instead of `await`ing the Promise right away and catching the exception, we pass the returned Promise directly to our test framework and say:

Please assert that this Promise will eventually reject with the following error.
We still need to `await` the whole thing because it is an async operation. (Not lazy, like in Rust but we want the test to wait until it is done).

Note that I also changed your assertion to actually check if we return the correct error by asserting the error message.

Reply:
A test purist would probably argue that you should even separate the assertion from the acting:

```ts
	// arrange
	const bodies = (await SwapFactory.newSwap(alice, bob))
                .hanEthereumEtherHalightLightningBitcoin;

	await alice.createSwap(bodies.alice);
	await sleep(200);

	// act
	let response = alice.createSwap(bodies.bob);

	// assert
	await expect(response).toThrow("Swap already exists.");
```

Reply:
Nice knowledge sharing, thanks for making the effort.  Will process and update the PR where it seems fit.